interchangenode/src/main/java/no/vegvesen/ixn/model/IxnMessage.java

```diff
@@ -38,6 +39,15 @@ public IxnMessage(TextMessage textMessage) throws JMSException {
         if (when != null) {
             this.when = when;
         }
+
+        Enumeration propertyNames = textMessage.getPropertyNames();
+        while (propertyNames.hasMoreElements()) {
+            String propertyName = (String) propertyNames.nextElement();
+            String propertyValue = textMessage.getStringProperty(propertyName);
```

Comment:
What if the property is not of the type String? e.g float or int.

Reply:
If we want to use int or float (or even boolean) properties, we can still represent them as strings over the wire, and parse them to the appropriate type internally. Any properties that should be treated as different datatypes have to be agreed in the protocol, though.